src/test/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScannerTest.java

```diff
@@ -406,6 +404,74 @@ public void testLeastCommonAncestor() throws Exception {
         Assert.assertEquals(2, starts.size());
     }
 
+    @Test
+    public void testVariousParallelCombos() throws Exception {
+        WorkflowJob job = r.jenkins.createProject(WorkflowJob.class, "ParallelTimingBug");
+        job.setDefinition(new CpsFlowDefinition(
+                "stage 'test' \n" +
+                        "    parallel([\n" +  // ID 5 is start
+                        "        'unit': {\n" +
+                        "          retry(1) {\n" +
+                        "            sleep 1;\n" +
+                        "            sleep 10; echo 'hello'; \n" +
+                        "          }\n" +
+                        "        },\n" +
+                        "        'otherunit': {\n" +
+                        "            retry(1) {\n" +
+                        "              sleep 1;\n" +
+                        "              sleep 5; \n" +
+                        "              echo 'goodbye'   \n" +
+                        "            }\n" +
+                        "        }\n" +  // end of branch:
+                        "    ])\n"
+        ));
+        /*Node dump follows, format:
```

Comment:
@jglick Yup, I hadn't forgotten it -- I think such a case might make a reasonable graph analysis feature too (not just as a test framework) -- unfortunately as we know there are lots of edge cases in graph structure present that need to be handled.  

I think it's not really viable until we have canonical tree-structured navigation of pipeline flow graphs -- which is the case that would benefit most from this for testing.